src/Data/StringVariants/NonEmptyText.hs

```diff
@@ -120,9 +121,12 @@ chunksOfNonEmptyText ::
   forall chunkSize totalSize.
   (KnownNat chunkSize, KnownNat totalSize, chunkSize <= totalSize, 1 <= chunkSize) =>
   NonEmptyText totalSize ->
-  [NonEmptyText chunkSize]
+  NE.NonEmpty (NonEmptyText chunkSize)
 chunksOfNonEmptyText (NonEmptyText t) =
-  mapMaybe mkNonEmptyText (T.chunksOf chunkSize t)
+  -- NE.fromList is partial. However, since we know the input
+  -- is nonempty, we're guaranteed to get at least one nonempty
+  -- chunk.
+  NE.fromList $ mapMaybe mkNonEmptyText (T.chunksOf chunkSize t)
```

Comment:
this could call `error` in the event that someone, say, misuses `unsafeMkNonEmptyText`. I don't know if we'd want a more descriptive exception?

Reply:
Yes, that'd be helpful